logstash-core/src/test/java/org/logstash/secret/cli/SecretStoreCliTest.java

```diff
@@ -387,6 +387,11 @@ private void assertNotListed(String... expected) {
         assertTrue(Arrays.stream(expected).noneMatch(terminal.out::contains));
     }
 
+    private String getRandomKeyName() {
+        final String[] uuidParts = UUID.randomUUID().toString().split("-");
+        return "r".concat(uuidParts[0]).concat("_").concat(uuidParts[1]);
```

Comment:
If were using UUID, it's easier to just replace the unsupported dashes ('-') with acceptable underscores, than to split and concat arbitrarily.

```suggestion
        return "r" + UUID.randomUUID().toString().replace('-','_');
```

Alternatively, we can generate the actual random keys that should be acceptable, which has the benefit of corroborating the requirement:

```suggestion
        final String firstCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_.";
        final String restCharacters = firstCharacters + "01234567890";

        final Random random = new Random();
        final StringBuilder sb = new StringBuilder();

        // add a random legal first-character
        sb.append(firstCharacters.charAt(random.nextInt(firstCharacters.length())));

        // add between 0 and 40 random legal rest-characters
        final int more = random.nextInt(40);
        random.ints(more, 0, restCharacters.length())
                .mapToObj(restCharacters::charAt)
                .forEach(sb::append);
        
        return sb.toString();
```